app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyPresenter.kt

```diff
@@ -52,11 +52,10 @@ class AppcoinsRewardsBuyPresenter(private val view: AppcoinsRewardsBuyView,
           rewardsManager.pay(transaction.skuId, amount, transaction.toAddress(), packageName,
               getOrigin(isBds, transaction), transaction.type, transaction.payload,
               transaction.callbackUrl, transaction.orderReference, transaction.referrerUrl)
-              .andThen(rewardsManager.getPaymentStatus(packageName, transaction.skuId,
-                  transaction.amount()))
+              .andThen(rewardsManager.getPaymentStatus(packageName, transaction.skuId, amount))
```

Comment:
We shouldn't be doing it this way, since the reference for the transaction is the transaction builder and it's what is being used on the other components and methods, for instance the events have in consideration the information on the transaction builder.

Also we need to make sure that the information in the transaction builder is the correct one, so I think we should after or update the transaction builder or update the URI that is being parsed